### PR TITLE
Add `global_shift` parameter to `PhasedISwapPowGate`

### PR DESCRIPTION
**Pull request by vtomole**
Fixes: https://github.com/quantumlib/Cirq/issues/4570

Follow up on review of https://github.com/quantumlib/Cirq/pull/4606

### CODE REVIEW

**Review by tanujkhattar — CHANGES_REQUESTED** [review]
We should also update, and add tests, for the following methods:

* `_decompose_` : Decompose preserving global phase.
* `_json_dict_` : Json serialization should also include global_shift. We should make sure deserialization of old objects still works. 
* Add tests for value equality of gates with non-zero global shift (which should work correctly because it depends upon `self._iswap._value_equality_values_`)
* `_with_exponent` should preserve the global_shift. 
* Add tests for pauli expansion with non-zero global shift

File: cirq-core/cirq/ops/phased_iswap_gate.py [hunk 1]
```diff
@@ -144,7 +156,7 @@ def _decompose_(self, qubits: Sequence['cirq.Qid']) -> 'cirq.OP_TREE':
 
         yield cirq.Z(a) ** self.phase_exponent
         yield cirq.Z(b) ** -self.phase_exponent
-        yield cirq.ISWAP(a, b) ** self.exponent
+        yield cirq.ISwapPowGate(global_shift=self.global_shift).on(a, b) ** self.exponent
```
Comment (vtomole) [inline cirq-core/cirq/ops/phased_iswap_gate.py]: > `_decompose_` : Decompose preserving global phase.
 
Done.

File: cirq-core/cirq/ops/phased_iswap_gate.py [hunk 2]
```diff
@@ -75,7 +78,11 @@ def _num_qubits_(self) -> int:
         return 2
 
     def _json_dict_(self) -> Dict[str, Any]:
```
Comment (vtomole) [inline cirq-core/cirq/ops/phased_iswap_gate.py:80]: >  `_json_dict_` : Json serialization should also include global_shift. We should make sure deserialization of old objects still works. 

Done.

File: cirq-core/cirq/ops/phased_iswap_gate_test.py [hunk 3]
```diff
@@ -96,28 +115,30 @@ def test_decompose_invalid_qubits():
 
 
 @pytest.mark.parametrize(
-    'phase_exponent, exponent',
+    'phase_exponent, exponent, global_shift',
```
Comment (vtomole) [inline cirq-core/cirq/ops/phased_iswap_gate_test.py:118]: > Add tests for pauli expansion with non-zero global shift

Done.

File: cirq-core/cirq/ops/phased_iswap_gate.py [hunk 4]
```diff
@@ -106,7 +113,9 @@ def _resolve_parameters_(
         )
 
     def _with_exponent(self, exponent: value.type_alias.TParamVal) -> 'PhasedISwapPowGate':
```
Comment (vtomole) [inline cirq-core/cirq/ops/phased_iswap_gate.py:115]: > `_with_exponent` should preserve the global_shift. 

Done.

File: cirq-core/cirq/ops/phased_iswap_gate_test.py [hunk 5]
```diff
@@ -25,19 +25,26 @@
...
+    s = 0.5
+    gate = cirq.PhasedISwapPowGate(phase_exponent=p, exponent=t, global_shift=s)
     assert gate.phase_exponent == p
     assert gate.exponent == t
+    assert gate.global_shift == s
 
 
 def test_phased_iswap_equality():
     assert cirq.PhasedISwapPowGate(phase_exponent=0, exponent=0.4) == cirq.ISWAP**0.4
+    assert (
```
Comment (vtomole) [inline cirq-core/cirq/ops/phased_iswap_gate_test.py:37]: > Add tests for value equality of gates with non-zero global shift (which should work correctly because it depends upon `self._iswap._value_equality_values_`)

Done.

Comment (tanujkhattar) [reply cirq-core/cirq/ops/phased_iswap_gate_test.py:37]: Also add a test for inequality
```
cirq.PhasedISwapPowGate(phase_exponent=0, exponent=0.4, global_shift=0.3) != cirq.ISWAP ** 0.4
```

It would be better to use EqualsTester if we can.

Comment (vtomole) [reply cirq-core/cirq/ops/phased_iswap_gate_test.py]: > It would be better to use EqualsTester if we can.

Done.

**Review by tanujkhattar — APPROVED** [review]
